classes/user_manager.php

```diff
@@ -207,6 +207,8 @@ public function upsert_user(array $data) {
                 throw new upset_failed_exception('error:invalidauth',  (object)['auth' => $auth]);
             }
 
+            $transaction = $DB->start_delegated_transaction();
```

Comment:
What was the real case for this issue? Can it be wrapped into a unit test?

Reply:
good point. Potentially this can happen when trying to upsert (create a new use) user with unique custom profile field, but the value already taken. 

Do you have any ideas or example on how to test this?

Reply:
I don't have examples, but feels like pre-creating a custom field, inserting a record to the table and then calling the function would work.

Reply:
Seems like this would take some time to implement proper unit tests (if even possible due to some restrictions around transactions in unit testing). Happy to merge this as is then